Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SchemaGeneration/RdbmsStructuredTypeDefinitionProviderTest.cs

```diff
@@ -0,0 +1,102 @@
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Data;
+using System.Linq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model.Building;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SchemaGeneration;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Model;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Model.Building;
+using Remotion.Development.UnitTesting;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SchemaGeneration
+{
+  [TestFixture]
+  public class RdbmsStructuredTypeDefinitionProviderTest
+  {
+    [Test]
+    public void GetTypeDefinitions ()
+    {
+      var provider = new RdbmsStructuredTypeDefinitionProvider();
+
+      var storageTypeInformationProvider = (IStorageTypeInformationProvider)new SqlStorageTypeInformationProvider();
+      var result = provider.GetTypeDefinitions(storageTypeInformationProvider).ToArray();
+
+      Assert.That(result.Length, Is.EqualTo(23));
+
+      var i = 0;
+      AssertFor(i++, DbType.String, typeof(String), false);
+      AssertFor(i++, DbType.AnsiString, typeof(String), false);
+      AssertFor(i++, DbType.Binary, typeof(Byte[]), false);
+      AssertFor(i++, DbType.Boolean, typeof(Boolean?), false);
+      AssertFor(i++, DbType.Boolean, typeof(Boolean?), true);
+      AssertFor(i++, DbType.Byte, typeof(Byte?), false);
+      AssertFor(i++, DbType.Byte, typeof(Byte?), true);
+      AssertFor(i++, DbType.DateTime, typeof(DateTime?), false);
+      AssertFor(i++, DbType.DateTime, typeof(DateTime?), true);
+      AssertFor(i++, DbType.Decimal, typeof(Decimal?), false);
+      AssertFor(i++, DbType.Decimal, typeof(Decimal?), true);
+      AssertFor(i++, DbType.Double, typeof(Double?), false);
+      AssertFor(i++, DbType.Double, typeof(Double?), true);
+      AssertFor(i++, DbType.Guid, typeof(Guid?), false);
+      AssertFor(i++, DbType.Guid, typeof(Guid?), true);
+      AssertFor(i++, DbType.Int16, typeof(Int16?), false);
+      AssertFor(i++, DbType.Int16, typeof(Int16?), true);
+      AssertFor(i++, DbType.Int32, typeof(Int32?), false);
+      AssertFor(i++, DbType.Int32, typeof(Int32?), true);
+      AssertFor(i++, DbType.Int64, typeof(Int64?), false);
+      AssertFor(i++, DbType.Int64, typeof(Int64?), true);
+      AssertFor(i++, DbType.Single, typeof(Single?), false);
+      AssertFor(i++, DbType.Single, typeof(Single?), true);
+      return;
```

Comment:
I would suggest dropping the return.  I understand it helps knowing the test is done and does not continue after the local function. Perhaps just turn it into a regular function?